test/Graphics/VertexArray.cpp

```diff
@@ -0,0 +1,88 @@
+#include <SFML/Graphics/VertexArray.hpp>
+#include "GraphicsUtil.hpp"
+
+#include <doctest.h>
+
+TEST_CASE("sf::VertexArray class - [graphics]")
+{
+    SUBCASE("Construction")
+    {
+        SUBCASE("Default constructor")
+        {
+            const sf::VertexArray vertexArray;
+            CHECK(vertexArray.getVertexCount() == 0);
+            CHECK(vertexArray.getPrimitiveType() == sf::PrimitiveType::Points);
+            CHECK(vertexArray.getBounds() == sf::FloatRect({0, 0}, {0, 0}));
+        }
+
+        SUBCASE("Explicit constructor with default argument")
+        {
+            const sf::VertexArray vertexArray(sf::PrimitiveType::Lines);
+            CHECK(vertexArray.getVertexCount() == 0);
+            CHECK(vertexArray.getPrimitiveType() == sf::PrimitiveType::Lines);
+            CHECK(vertexArray.getBounds() == sf::FloatRect({0, 0}, {0, 0}));
+        }
+
+        SUBCASE("Explicit constructor")
+        {
+            const sf::VertexArray vertexArray(sf::PrimitiveType::Lines, 10);
+            CHECK(vertexArray.getVertexCount() == 10);
+            CHECK(vertexArray.getPrimitiveType() == sf::PrimitiveType::Lines);
+            CHECK(vertexArray.getBounds() == sf::FloatRect({0, 0}, {0, 0}));
+            for (std::size_t i = 0; i < vertexArray.getVertexCount(); ++i)
+            {
+                CHECK(vertexArray[i].position == sf::Vertex().position);
+                CHECK(vertexArray[i].color == sf::Vertex().color);
+                CHECK(vertexArray[i].texCoords == sf::Vertex().texCoords);
+            }
+        }
+    }
+
+    SUBCASE("Resize array")
+    {
+        sf::VertexArray vertexArray;
+        vertexArray.resize(42);
+        CHECK(vertexArray.getVertexCount() == 42);
```

Comment:
Should we check the value of an inserted element (e.g. at index 41)?

Reply:
The explicit constructor tests already test that, I believe. I loop through all the vertices and make sure their values match that of a default constructed `sf::Vertex`.

Reply:
Yes, but the constructor and `resize()` are two separate APIs.

Maybe we can extract a function for "compare equality of `position`, `color` and `texCoords`".

Reply:
Fair. I just realized that the explicit constructor isn't actually calling `resize()` in the implementation. I added those checks to this test.

Reply:
Even if it were calling `resize()` internally -- strictly speaking, this is an implementation detail that can change at any time and is exactly the reason why we have tests 🙂 

(We have quite someconfidence that the _current_ implementation works, it's more if anything would ever change. Rather unlikely in this case though.)